### ISSUE TITLE
Custom Templates

### ISSUE THREAD

**Issue by AlexDenisov** [danger/danger#1101]
Not really a report, but rather a question/suggestion.

I am considering adding custom, user-provided templates for a `Dangerfile`, e.g.:
```
danger --template foobar.erb
```
or
```ruby
# Dangerfile
template 'foobar.erb'
```
or in some other way.

I looked at the source code and it looks doable without much effort except of one thing: it is very likely that Danger will not be able to update or remove previous comment if the user-provided template does not have the `danger_id` included or errors/warnings/messages will have a different structure.

This issue does not concern my use case because we do not update/remove previous comments in order to keep the 'build history' visible and persistent, but it can definitely be show stopper for Danger.

To conclude: would you accept a feature that works only if a user is careful enough?

I am happy to elaborate more if my message is not clear, so please do not hesitate to ask questions.

**Comment by orta** [comment]
Yep, I get what you mean - seems reasonable to me (so long as it's well tested and doesn't seem to break the default user flow ) 

There might be a way to make the main flow work that way (we already have to have different templates for GitHub vs gitlab for example, and this could consolidate the logic) which uses a default template instead of whatever it does now

**Comment by AlexDenisov** [comment]
Sounds great, @orta. I've seen some duplication there as well, so this feature may require some ground work before.

I will do some more digging into the matter to see how to approach it and will get back to you then, hopefully with some plan or pull requests :)

**Comment by AlexDenisov** [comment]
Just wanted to mention that our internal priorities have changed and I won't be able to get back to this feature in the near future.
Thank you for understanding.
